src/public.js

```diff
@@ -42,7 +49,11 @@ window.addEventListener('DOMContentLoaded', function() {
 
 		const sharingToken = sharingTokenElmt.value
 		const downloadUrl = generateUrl('/s/{token}/download', { token: sharingToken })
-		const viewerUrl = generateUrl('/apps/files_pdfviewer/?file={downloadUrl}#page={page}', { downloadUrl, page })
+		const viewerUrl = generateUrl('/apps/files_pdfviewer/?file={downloadUrl}&canDownload={canDownload}#page={page}', {
+			canDownload: canDownload() ? 1 : 0,
+			downloadUrl,
+			page,
+		})
 
 		// Create viewer frame
 		const viewerNode = document.createElement('iframe')
```

Comment:
Looking at the `<iframe>` specs, maybe the `sandbox="allow-downloads"` attribute could be useful. But I am not sure that it restricts every kind of download:

> allow-downloads: Allows for downloads to occur with a gesture from the user.

If it really does restrict every download, it would be a cleaner way to forward `canDownload`.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe

Reply:
Arf, Safari does not support it...

Reply:
Well, there are wider things to manage than downloading. Printing for exampel and selecting the text. So i'd rather have one implementation to rule them all :ring: